MIGRATION.md

```diff
@@ -32,6 +32,42 @@ import {
 } from '@sentry/minimal';
 ```
 
+## Explicit Client Options
+
+In v7, we've updated the `Client` to have options seperate from the options passed into `Sentry.init`. This means that constructing a client now requires 3 options: `integrations`, `transport` and `stackParser`. These can be customized as you see fit.
+
+```ts
+import { BrowserClient, defaultStackParser, defaultIntegrations, makeFetchTransport } from '@sentry/browser';
+
+// New in v7:
+const client = new BrowserClient({
+  transport: makeFetchTransport,
+  stackParser: defaultStackParser,
+  integrations: [...defaultIntegrations],
+});
+
+// Before:
+const client = new BrowserClient();
+```
+
+Since you now explicitly pass in the dependencies of the client, you can also tree-shake out dependencies that you do not use this way. For example, you can tree-shake out the SDK's default integrations and only use the ones that you want like so:
+
+```ts
+import { BrowserClient, defaultStackParser, Integrations, makeFetchTransport } from '@sentry/browser';
+
+// New in v7:
+const client = new BrowserClient({
+  transport: makeFetchTransport,
+  stackParser: defaultStackParser,
+  integrations: [
+    new Integrations.Breadcrumbs(),
+    new Integrations.GlobalHandlers(),
+    new Integrations.LinkedErrors(),
+    new Integrations.Dedupe(),
+  ],
```

Comment:
Is this style of exporting the integrations tree-shakeable?

Wouldn't be better to do:

```ts
import { BreadcrumbsIntegration } from '@sentry/browser';
```

Reply:
You're right actually! This doesn't treeshake because we are constructing an object with spreads and then exporting it.

https://github.com/getsentry/sentry-javascript/blob/39553b56cd1162db13e9cc512988728f01d9dca6/packages/browser/src/index.ts#L16-L20

We'll have to switch to exporting everything individually. Will take care of that!

Reply:
For now just imported everything individually, we can think about improving the structure at another time: https://github.com/getsentry/sentry-javascript/pull/5028